src/components/pagination/bl-pagination.ts

```diff
@@ -0,0 +1,239 @@
+   */
+  @property({ attribute: 'select-label', type: String })
+  selectLabel = 'Show';
+
+  /**
+   *  Sets the option texts.
+   */
+  @property({ attribute: 'option-text', type: String })
+  optionText = 'Items';
+
+  @state() private pages: Array<number | string> = [];
+
+  /**
+   * Fires when the current page changes
+   */
+  @event('bl-change') private onChange: EventDispatcher<{ selectedPage: number; prevPage: number }>;
+
+  updated(changedProperties: PropertyValues<this>) {
+    if (changedProperties.has('currentPage') || changedProperties.has('itemsPerPage')) {
+      this._paginate();
+      this.onChange({
+        selectedPage: this.currentPage,
+        prevPage: changedProperties.get('currentPage'),
+      });
+    }
+  }
+
+  private _paginate() {
+    this.pages = [];
+    const pageListLength = Math.ceil(Math.abs(this.totalItems / this.itemsPerPage)) || 1;
+
+    if (pageListLength <= 8) {
+      this.pages = Array.from(Array(pageListLength), (_, index) => index + 1);
+      return;
+    }
+
+    this.pages.push(1);
+
+    if (this.currentPage < 5) {
+      this.pages.push(2, 3, 4, 5, '...');
+    } else if (this.currentPage >= 5 && this.currentPage <= pageListLength - 4) {
+      this.pages.push('...', this.currentPage - 1, this.currentPage, this.currentPage + 1, '...');
+    } else {
+      this.pages.push(
+        '...',
+        pageListLength - 4,
+        pageListLength - 3,
+        pageListLength - 2,
+        pageListLength - 1
+      );
+    }
+
+    this.pages.push(pageListLength);
+  }
+
+  private _changePage(page: number): void {
+    this.currentPage = page;
+  }
+
+  private _pageBack(): void {
+    if (this.currentPage === 1) return;
+    this.currentPage--;
+  }
+
+  private _pageForward(): void {
+    if (this.currentPage === this._getLastPage()) return;
+    this.currentPage++;
+  }
+
+  private _getLastPage(): number {
+    return +this.pages[this.pages.length - 1];
+  }
+
+  private _inputHandler(event: CustomEvent) {
+    const inputValue = +(event.target as HTMLInputElement).value;
+    const newPage = inputValue > 0 ? Math.min(this._getLastPage(), inputValue) : 1;
+    this._changePage(newPage);
+  }
+
+  private _selectHandler(event: CustomEvent) {
+    this.itemsPerPage = event?.detail[0]?.value;
+    this.currentPage = 1;
+  }
+
+  private renderSinglePage(page: number | string) {
+    if (typeof page === 'string') {
+      return html`<span class="dots"></span>`;
+    }
+    return html` <li>
+      <bl-button
+        @click="${() => this._changePage(page)}"
+        variant="secondary"
+        kind=${this.currentPage === page ? 'contained' : 'text'}
+      >
+        ${page}
+      </bl-button>
+    </li>`;
+  }
+
+  private renderPages() {
+    return html`
+      <div class="page-container">
+        <bl-button
+          @click="${this._pageBack}"
+          kind="text"
+          variant="secondary"
+          icon="arrow_left"
+          class="arrow-left"
```

Comment:
I would use a class name like `previous` instead of the current visual situation.